Gestione autoparco

ico88 opened the pull request:
Implementazione dell'autoparco:
- inserimento veicoli (targa, tipologia, selettiva radio)
- modifica veicoli
- gestione a db delle tipologie veicoli (inserimento e modifica)

Commit:
Creazione Modulo Autoparco, gestione permessi di accesso a modulo

Commit:
Creazione ripeilogo veicoli, pagina di inserimento veicoli, correzione e accentate

biagiosaitta committed:
Corretto Inserimento Veicoli

Commit:
Creazione pagine di modifica veicolo

Commit:
Normalizzazione targa e selttiva nei veicoli

Commit:
Tipologia dei veicoli in DB close #14

AlfioEmanueleFresta commented on .gitignore:
```diff
@@ -28,4 +28,6 @@ public/images/barcode/*
\ No newline at end of file
+/public/images/intestazione_certificato.png
+
+vendor/
```
Se fai un `git pull origin master` per tirare le ultime modifiche da `master` dentro il tuo branch, troverai che nell'ultima versione del `.gitignore` (dopo le mie modifiche) è presente la cartella `.idea/` (generata automaticamente da PhpStorm), così che non venga accidentalmente inclusa automaticamente nei commit.

AlfioEmanueleFresta commented on public/class/Tipologia-veicolo.obj.inc:
```diff
@@ -0,0 +1,66 @@
+
+require_once('../db-connessione-include.php');
+
+class Tipologia_veicolo{
```
Ti suggerirei di usare una nomenclatura  `CamelCase` per le classi, per consistenza con il resto del codice, i.e. `TipologiaVeicolo`. PhpStorm ti permette di rinominare facilmente una classe, andando a cambiare tutte le occorrenze del nome automaticamente per te (click destro > refactor > rename).

AlfioEmanueleFresta commented on public/livesearch-autoparco-tipologie.js:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    return null;
+}
```
`XmlHttpObject` e' davvero vecchio e deprecato. Tutto il tuo codice in questo file puo' essere sostituito con una versione equivalente, ma piu' moderna e breve, come segue:

```javascript
async function showResult(str, num) {
    var risposta = await fetch(`livesearch-autoparco-tipologie.php?q=${str}&num=${num}`);
    var html = await risposta.text();
    document.getElementById("livesearch").innerHTML = html;
}
```

AlfioEmanueleFresta commented on public/livesearch-autoparco-tipologie.php:
```diff
@@ -0,0 +1,55 @@
+    include '../db-connessione-include.php';
+    include 'maledetti-apici-centro-include.php'; //ATTIVA O DISATTIVA IL MAGIC QUOTE PER GLI APICI
+
+    function __autoload ($class_name) { //funzione predefinita che si occupa di caricare dinamicamente tutti gli oggetti esterni quando vengono richiamati
```
Fai `git pull origin master` per tirarti le ultime modifiche che sono già in master, e rimuovi poi del tutto la funzione `__autoload` (righe 12-14), perché non è più necessaria e rompe tutto il nuovo codice, come @biagiosaitta ha già sperimentato (AWS).

AlfioEmanueleFresta replied:
Tieni presente che il file `db-connessione-include.php` si chiama ora `core.php` (perché non contiene solamente roba relativa al database). Dopo aver tirato le ultime modifiche, dovrai cambiare tutte le occorrenze da

```php
include '../db-connessione-include.php';
```

a

```php
include '../core.php';
```

AlfioEmanueleFresta commented on public/livesearch-autoparco-tipologie.php:
```diff
@@ -0,0 +1,55 @@
+    $ogg = $_GET['q'];
+    $num = $_GET['num'];
+
+    $query = $connessione->query("SELECT * FROM aut_tipologie WHERE descrizione LIKE '%$ogg%' ORDER BY descrizione ASC LIMIT $num");
```
Questo codice è vulnerabile ad attacchi di tipo SQL injection.

Non usare *mai* variabili direttamente in una query. Piuttosto, usa sempre i *prepared statements* di PDO, come segue:

```php
$query = $connessione->prepare("SELECT * FROM aut_tipologie WHERE descrizione LIKE :descrizione ORDER BY descrizione ASC LIMIT :num");
$query->bindValue(":descrizione", "%" . $ogg . "%");
$query->bindValue(":num", $num, PDO::PARAM_INT);
$query->execute();
```

Commit:
Correzione Typo e inserimento libretto veicoli

Commit:
Creazione classe dedicata veicoli, inserita cancellazione veicoli

Commit:
Correzione messaggio di conferma durante la cancellazione dei veicoli

Commit:
Closed #18 classe veicolo
Inserimento ruoli Op CO e Resp CO, creazione pagina di gestione volontari autorizzati

Commit:
Gestione lista volontari, prima stesura

Commit:
Gestione lista volontari ref #22, sistemazione sessione e identazione, creazione pulsante aggiunta autorizzati
